### ISSUE TITLE
Preview API response on collection/document view

### ISSUE THREAD

**Issue by Kematia** [directus/app-next#770]
Hey! 

Just a little feature request: I'd love to have a button/link that opens a new window with the API response for the current document/collection I'm looking at. It would improve the speed for prototyping the API and maybe for passing it around to other devs.

One tricky thing could be the permissions, but just having a preview on the public role could be cool. Maybe we could pass the session/token/whatisit of the current user in the URL?

And also the depth (the "fields: *.*.*" parameter iirc from the JS sdk). 

I could work on it if needed! 
Keen to hear your feedback

**Comment by benhaynes** [comment]
Absolutely, this is something that I wanted for a while, and I believe there might be another ticket or feature request here somewhere for it. The way I see it is there's a secondary button somewhere on the page that will expose the API URLs used, and potentially open to show the raw JSON of the page. This ties into our import export feature request.

**Comment by benhaynes** [comment]
Also, we'd love the help! That we'd probably want to coordinate with Rijk to work on the version 9 codebase, probably on the node port (unless you're more comfortable on PHP).

**Comment by benhaynes** [comment]
Also, we'd love the help!!

**Comment by Kematia** [comment]
I'd love to! I'll give a look at the codebase over the weekend. 

Do you reckon is something we can implement in the App only or does it need API work too? Having a "tab" open inside the current document/collection view (and browser window) could be easier for authentication and filtering? IE: use the current session/token to authenticate it, instead of handing off everything to a new browser window?

**Comment by benhaynes** [comment]
I'll defer to @rijkvanzanten on that, but that makes sense to me. I think this should be all app since it's JSON output. It'd be nice if there was an option to display it with or without formatting... but again I defer to Rijk.

**Comment by rijkvanzanten** [comment]
@benhaynes @Kematia Would it be an idea to have an admin only section under the help and docs section in the sidebar that shows the API endpoint + filters used to get the current view?

@Kematia FYI Directus v9 has this modular sidebar setup:

![CleanShot 2020-07-07 at 14 00 23](https://user-images.githubusercontent.com/9141017/86822943-3d85f400-c05a-11ea-8220-04bfd1659961.png)

🙂

**Comment by rijkvanzanten** [comment]
Very _very_ rough quick and dirty hacked in example:

![CleanShot 2020-07-07 at 14 04 08](https://user-images.githubusercontent.com/9141017/86823358-c1d87700-c05a-11ea-9ad2-3fae97914ab6.png)

**Comment by benhaynes** [comment]
I like that! Or can you even be a dedicated component...

**Comment by Kematia** [comment]
Love the mockup!
Yeah, the sidebar could be a perfect place for it, even tho I'd love to have a modal/wider page view... the sidebar could be the place to costumize the request (maybe just public or current signed in user for simplicity sake, and the fields' depth)

@benhaynes  Isn't there a JSON field component already? The sidebar would just set up the request (with maybe a couple additional parameter fields, like the aforementioned role-picker) and populate it

**Comment by rijkvanzanten** [comment]
I'm wondering how much of this is "Open current view in API" vs dynamic API reference. We've been talking about creating / including a dynamic API reference with code examples based on your current schema, which would achieve a similar goal I think

**Comment by benhaynes** [comment]
I think both are useful. I think having the JSON there with the specific on page parameters and API requests is really useful. And if/when we have a dynamic API reference built in, we can link from the general endpoint to that section of the docs... maybe even highlighting the chosen params.

**Comment by Kematia** [comment]
I think I'll work a bit on the "Open current view in API" concept for now! I'll have a look at the codebase in this weekend and see if I can work with existing components 😄

**Comment by rijkvanzanten** [comment]
Before you dive in to doing too much work, lets figure out if this is something we want to implement in the core codebase (80/20) or if it should be an extension 🙂 

While I think it's an interesting concept, I'm not fully convinced it passes the 80/20 rule. I wanna make sure we don't end up in a massive feature bloat 😄

**Comment by benhaynes** [comment]
While I agree that we need to remain diligent in terms of our 80/20 rule, I believe this feature passes that... and I'd be very excited to include it. I think the key here is to ensure that this doesn't busy up the UI for non-developers, so I think our focus should be on scoping this to the admin role, or somehow enabling it through permissions.

I'm fine if we want to make this is't own modular sidebar component, but we haven't figured out how to make those extensions yet, and I would want this feature to ship with Directus Core OOTB anyway.

**Comment by Kematia** [comment]
I don't mind doing some work even if it's not accepted, because:

1. I'm interested in getting to know the v9 codebase, because I will probably need to work on it for future clients
2. I'd like to see a bigger vue app is structured
3. if it's not in the Core I'll probably add it anyway as a custom plugin when we use Directus in-house 😄
